crates/turbopack-ecmascript/src/analyzer/mod.rs

```diff
@@ -211,6 +289,25 @@ impl LogicalOperator {
     }
 }
 
+#[derive(Debug, Clone, Copy, Hash, PartialEq, Eq)]
+pub enum BinaryOperator {
+    Equal,
+    NotEqual,
+    StrictEqual,
+    StrictNotEqual,
+}
+
+impl BinaryOperator {
+    fn joiner(&self) -> &'static str {
+        match self {
+            BinaryOperator::Equal => " == ",
+            BinaryOperator::NotEqual => " != ",
+            BinaryOperator::StrictEqual => " === ",
+            BinaryOperator::StrictNotEqual => " !== ",
+        }
```

Comment:
Looks like there's already a space in:

```rust
            JsValue::Binary(_, a, op, b) => write!(f, "({} {} {})", a, op.joiner(), b),
```

so these should not have leading and trailing spaces.